pkg/destroy/openstack/openstack.go

```diff
@@ -655,27 +656,29 @@ func deleteContainers(opts *clientconfig.ClientOpts, filter Filter, logger logru
 			// Openshiftclusterid in the X-Container-Meta- HEAD output
 			titlekey := strings.Title(strings.ToLower(key))
 			if metadata[titlekey] == val {
-				listOpts := objects.ListOpts{Full: false}
-				allPages, err := objects.List(conn, container, listOpts).AllPages()
-				if err != nil {
-					logger.Error(err)
-					return false, nil
-				}
-				allObjects, err := objects.ExtractNames(allPages)
-				if err != nil {
-					logger.Error(err)
-					return false, nil
-				}
-				for _, object := range allObjects {
-					logger.Debugf("Deleting object %q", object)
-					_, err = objects.Delete(conn, container, object, nil).Extract()
+				logger.Debugf("Bulk deleting container %q objects", container)
+				pager := objects.List(conn, container, &objects.ListOpts{
+					Limit: 50,
+				})
+				err = pager.EachPage(func(page pagination.Page) (bool, error) {
+					objectsOnPage, err := objects.ExtractNames(page)
 					if err != nil {
-						// Ignore the error if the object cannot be found and return with an appropriate message if it's another type of error
-						if _, ok := err.(gophercloud.ErrDefault404); !ok {
-							logger.Errorf("Removing object %q from container %q failed: %v", object, container, err)
-							return false, nil
-						}
-						logger.Debugf("Cannot find object %q in container %q. It's probably already been deleted.", object, container)
+						return false, err
+					}
+					resp, err := objects.BulkDelete(conn, container, objectsOnPage).Extract()
+					if err != nil {
+						return false, err
+					}
+					if len(resp.Errors) > 0 {
+						return false, errors.Errorf("Errors occured during bulk deleting of container %q objects", container)
```

Comment:
errors shoud start with capital letters as it makes it weird when wrapped

Reply:
This is hiding the actaul errorrs in the returned error. You can try and use https://godoc.org/k8s.io/apimachinery/pkg/util/errors#NewAggregate if the list of `resp.Errors` implements error iterface

Reply:
done!